src/game/client/components/menus_demo.cpp

```diff
@@ -569,13 +570,16 @@ void CMenus::RenderDemoList(CUIRect MainView)
 				}
 				else if(ID == COL_DEMO_LENGTH && !r.front().m_IsDir && r.front().m_InfosLoaded)
 				{
-					int Length = r.front().Length();
-					char aLength[32];
-					str_format(aLength, sizeof(aLength), "%d:%02d", Length/60, Length%60);
-					Button.VMargin(4.0f, &Button);
-					if(!Item.m_Selected)
-						TextRender()->TextColor(CUI::ms_TransparentTextColor);
-					UI()->DoLabel(&Button, aLength, Item.m_Rect.h*ms_FontmodHeight*0.8f, CUI::ALIGN_LEFT);
+					if (DemoItem.m_Valid)
```

Comment:
Could add this `if` check to the `else if` in line 571 as well. Same for the below `else if`.

Also this method uses `r.front()` multiple times instead of the existing `DemoItem` variable.